### PR TITLE
importer: generate string by stats

### PR DESCRIPTION
**Pull request by hanfei1991**
@fipped PTAL

### CODE REVIEW

**Comment by coocood** [comment]
@hanfei1991 
The length of the string may be much less than the defined max length.
But we have the stats, we can infer the average string length by the bucket values.

File: cmd/importer/db.go [hunk 1]
```diff
@@ -46,6 +46,9 @@ func intRangeValue(column *column, min int64, max int64) (int64, int64) {
 }
 
 func randStringValue(column *column, n int) string {
+	if column.hist != nil {
+		return column.hist.randString(n)
```
Comment (fipped) [inline cmd/importer/db.go]: Here _n_ is the maximum length of the rand string, but `column.hist.randString(n)` seems to return the rand string with length _n_ exactly.

File: cmd/importer/stats.go [hunk 2]
```diff
@@ -83,3 +84,37 @@ func (h *histogram) randInt() int64 {
...
 	return h.Bounds.GetRow(idx).GetInt64(0)
 }
+
+func getValidPrefix(lower, upper string) string {
+	prefixLen := 0
+	for i := range lower {
+		if i >= len(upper) {
+			log.Fatal("lower %s is larger than upper %s", lower, upper)
+		}
+		if lower[i] != upper[i] {
```
[the remark this answers is not in the document]
Comment (hanfei1991) [reply cmd/importer/stats.go]: No, I think the json file is valid

**Comment by hanfei1991** [comment]
@coocood The average length may not be much less. We use lower and upper bound to get a prefix length, and random a total length. This is close to `max length / 2`

**Comment by coocood** [comment]
@hanfei1991 
The avarage length of real values may be much less than `max length / 2`.

**Comment by hanfei1991** [comment]
@coocood after we have stats of average length, we can do it.

**Comment by coocood** [comment]
LGTM

File: cmd/importer/stats.go [hunk 3]
```diff
@@ -83,3 +85,51 @@ func (h *histogram) randInt() int64 {
...
+	return lower
+}
+
+func (h *histogram) getAvgLen(maxLen int) int {
+	l := h.Bounds.NumRows()
+	totalLen := 0
+	for i := 0; i < l; i++ {
+		totalLen += len(h.Bounds.GetRow(i).GetString(0))
+	}
+	avg := totalLen / l
```
Comment (lamxTyler) [inline cmd/importer/stats.go]: Is it possible that `l` is 0?

File: cmd/importer/stats.go [hunk 4]
```diff
@@ -83,3 +85,51 @@ func (h *histogram) randInt() int64 {
...
+	idx := h.getRandomBoundIdx()
+	if idx%2 == 0 {
+		lower := h.Bounds.GetRow(idx).GetString(0)
+		upper := h.Bounds.GetRow(idx + 1).GetString(0)
+		prefix := getValidPrefix(lower, upper)
+		restLen := h.avgLen - len(prefix)
+		if restLen > 0 {
+			prefix = prefix + randString(restLen)
+		}
+		log.Warnf("prefix %s", prefix)
```
Comment (lamxTyler) [inline cmd/importer/stats.go]: What's the propose of this log?

File: cmd/importer/stats.go (hunk 3 above)
Comment (hanfei1991) [reply cmd/importer/stats.go]: no, If length of bucket is equal to 0, we will not set the histogram.

File: cmd/importer/stats.go (hunk 4 above)
Comment (hanfei1991) [reply cmd/importer/stats.go]: My fault

**Pull request merged**